### PR TITLE
PS: More fine-grained file indexing support

### PR DESCRIPTION
**Pull request by MathiasVP**
This PR adds two mostly-separate features to the Powershell extractor:

1. The first commit adds support for the `--file-list` command-line option in the extractor. This was the easiest way to support running `codeql test run` which supplies a path to a newline-separated list of files to index. `--file-list` reads this file and indexes the files listed.
2. The second commit adds support for the `LGTM_INDEX_INCLUDE` environment variable similar to how Python and JS handles it: If `LGTM_INDEX_INCLUDE` is set to a semicolon-separated list of paths, this list of paths instead of every `.ps1` file in the current directory.

I apologize for the changes to formatting. It seems like I'm running a different formatter than the one used in the rest of the file. I recommend reviewing this with whitespace ignored!

### CODE REVIEW

File: powershell/tools/qltest.cmd [hunk 1]
```diff
@@ -1,9 +1,11 @@
 @echo off
 
-if not defined CODEQL_POWERSHELL_EXTRACTOR (
-    set CODEQL_POWERSHELL_EXTRACTOR=Semmle.Extraction.PowerShell.Standalone.exe
-)
-
-type NUL && "%CODEQL_EXTRACTOR_POWERSHELL_ROOT%/tools/%CODEQL_PLATFORM%/%CODEQL_POWERSHELL_EXTRACTOR%"
+type NUL && "%CODEQL_DIST%\codeql.exe" database index-files ^
+    --prune=**/*.testproj ^
+    --include-extension=.ps1 ^
```
Comment (dilanbhalla) [inline powershell/tools/qltest.cmd:5]: Anything other extensions we should consider, like psm1 and/or psd1?

File: powershell/extractor/Semmle.Extraction.PowerShell.Standalone/Options.cs [hunk 2]
```diff
@@ -70,12 +79,48 @@ public override void InvalidArgument(string argument)
...
+        /// <summary>
+        /// Returns a list of files to extract. By default, this is the list of all files in
+        /// the current directory. However, if the LGTM_INDEX_INCLUDE environment variable is
+        /// set, it is used as a list of files to include instead of the files from the current
+        /// directory.
+        /// </summary>
+        private static FileInfo[] GetDefaultFiles()
+        {
+            // Check if the LGTM_INDEX_INCLUDE environmen variable exists
+            var include = System.Environment.GetEnvironmentVariable("LGTM_INDEX_INCLUDE");
```
Comment (dilanbhalla) [inline powershell/extractor/Semmle.Extraction.PowerShell.Standalone/Options.cs:105]: This is great. Reminds me that there is LGTM_INDEX_EXCLUDES as well, but we can write that up once there is an ask for it

**Review by dilanbhalla — APPROVED** [review]

File: powershell/tools/qltest.cmd [hunk 3]
```diff
@@ -1,9 +1,11 @@
...
+    --size-limit=5m ^
+    --language=powershell ^
+    --working-dir=. ^
+    "%CODEQL_EXTRACTOR_POWERSHELL_WIP_DATABASE%"
```
Comment (dilanbhalla) [inline powershell/tools/qltest.cmd:9]: Doesn't this need to be a list of files? I guess the CI works so I must be missing something

File: powershell/tools/index-files.sh [hunk 4]
```diff
@@ -4,4 +4,4 @@ if [[ -z "${CODEQL_EXTRACTOR_POWERSHELL_ROOT}" ]]; then
   export CODEQL_EXTRACTOR_POWERSHELL_ROOT="Semmle.Extraction.PowerShell.Standalone.exe"
 fi
 
-"$CODEQL_EXTRACTOR_POWERSHELL_ROOT/tools/$CODEQL_PLATFORM/$CODEQL_POWERSHELL_EXTRACTOR"
\ No newline at end of file
+"$CODEQL_EXTRACTOR_POWERSHELL_ROOT/tools/$CODEQL_PLATFORM/$CODEQL_POWERSHELL_EXTRACTOR" --file-list "%1"
```
Comment (dilanbhalla) [inline powershell/tools/index-files.sh:7]: I know `codeql test run` passes a list of files as an argument here, but what about other commands that call this script? Does this make it always necessary to pass a file list (as opposed to a directory)? My bad for the rookie questions here, just want to ensure I am understanding the extractor as best as possible.

File: powershell/tools/qltest.cmd (hunk 3 above)
Comment (MathiasVP) [reply powershell/tools/qltest.cmd:9]: Note that this calls `codeql database index-files`. That doesn't take a list of files. That command will ultimately end up calling the `index-files` script with an argument that is a path to a file containing a newline-separated list of files to index.

File: powershell/tools/qltest.cmd (hunk 1 above)
Comment (MathiasVP) [reply powershell/tools/qltest.cmd:5]: Yeah, I think all of those are valuable to extract. I don't know if the current extractor supports them yet, though: https://github.com/microsoft/codeql/blob/22a30ab952783e6c555ce852bce7daebe12ceacb/powershell/extractor/Semmle.Extraction.PowerShell.Standalone/Options.cs#L68

We should add extraction of these other files as well, though

File: powershell/tools/index-files.sh (hunk 4 above)
Comment (MathiasVP) [reply powershell/tools/index-files.sh:7]: I think it always makes sense. For example, this is what Ruby does as well: https://github.com/github/codeql/blob/main/ruby/tools/index-files.cmd. So I think we're good 🤞

File: powershell/extractor/Semmle.Extraction.PowerShell.Standalone/Options.cs (hunk 2 above)
Comment (MathiasVP) [reply powershell/extractor/Semmle.Extraction.PowerShell.Standalone/Options.cs:105]: Indeed, that's a logical next step 😄

**Pull request merged**